app/apollo/resolvers/group.js

```diff
@@ -485,18 +529,14 @@ const groupResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, groupUuids, clusterId }, `${queryName} validating`);
-
-        await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context);
-
         validateString( 'org_id', org_id );
         validateString( 'clusterId', clusterId );
         groupUuids.forEach( value => validateString( 'groupUuids', value ) );
 
-        const groups = await models.Group.find({ org_id, uuid: { $in: groupUuids } });
-        if (groups.length != groupUuids.length) {
-          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
-        }
+        let groups = await getAllowedResources(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, null, groupUuids);
+        logger.info({req_id, user, org_id, clusterId, groupUuids}, `${queryName} retrieved allowed resources`);
+
+        if (groups.length < 1) { throw new NotFoundError(context.req.t('None of the passed group uuids were found')); }
```

Comment:
Removing the check for 'groups.length != groupUuids.length` would be incorrect -- if the API input is "make this cluster a member of these 5 groups" and only one of those groups is found, it should fail, not just set that one group.

Reply:
`groups` variable should not be re-assignable:
```suggestion
        const groups = await getAllowedResources(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, null, groupUuids);
        logger.info({req_id, user, org_id, clusterId, groupUuids}, `${queryName} retrieved allowed resources`);

        if (groups.length < 1) { throw new NotFoundError(context.req.t('None of the passed group uuids were found')); }
```

Reply:
fixed